synfig-studio/src/synfigapp/actions/layerlowermax.cpp

```diff
@@ -0,0 +1,157 @@
+/* === S Y N F I G ========================================================= */
+/*!	\file layerlowermax.cpp
+**	\brief Template File
+**
+**	\legal
+**	Copyright (c) 2002-2005 Robert B. Quattlebaum Jr., Adrian Bentley
+**	Copyright (c) 2008 Chris Moore
+**
+**	This file is part of Synfig.
+**
+**	Synfig is free software: you can redistribute it and/or modify
+**	it under the terms of the GNU General Public License as published by
+**	the Free Software Foundation, either version 2 of the License, or
+**	(at your option) any later version.
+**
+**	Synfig is distributed in the hope that it will be useful,
+**	but WITHOUT ANY WARRANTY; without even the implied warranty of
+**	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+**	GNU General Public License for more details.
+**
+**	You should have received a copy of the GNU General Public License
+**	along with Synfig.  If not, see <https://www.gnu.org/licenses/>.
+**	\endlegal
+*/
+/* ========================================================================= */
+
+/* === H E A D E R S ======================================================= */
+#ifdef USING_PCH
+#	include "pch.h"
+#else
+#ifdef HAVE_CONFIG_H
+#	include <config.h>
+#endif
+
+#include <synfig/general.h>
+
+#include "layerlowermax.h"
+#include "layermove.h"
+#include <synfigapp/canvasinterface.h>
+
+#include <synfigapp/localization.h>
+
+#endif
+
+using namespace synfig;
+using namespace synfigapp;
+using namespace Action;
+
+/* === M A C R O S ========================================================= */
+
+ACTION_INIT_NO_GET_LOCAL_NAME(Action::LayerLowerMax);
+ACTION_SET_NAME(Action::LayerLowerMax,"LayerLowerMax");
+ACTION_SET_LOCAL_NAME(Action::LayerLowerMax,N_("Lower Layer to Bottom"));
+ACTION_SET_TASK(Action::LayerLowerMax,"lower to bottom");
+ACTION_SET_CATEGORY(Action::LayerLowerMax,Action::CATEGORY_LAYER);
+ACTION_SET_PRIORITY(Action::LayerLowerMax,9);
+ACTION_SET_VERSION(Action::LayerLowerMax,"0.0");
+
+/* === G L O B A L S ======================================================= */
+
+/* === P R O C E D U R E S ================================================= */
+
+/* === M E T H O D S ======================================================= */
+
+LayerLowerMax::LayerLowerMax()
+{
+
+}
+
+synfig::String
+Action::LayerLowerMax::get_local_name()const
+{
+	return get_layer_descriptions(layers, _("Lower Layer to Bottom"), _("Lower Layers to Bottom"));
+}
+
+Action::ParamVocab
+Action::LayerLowerMax::get_param_vocab()
+{
+	ParamVocab ret(Action::CanvasSpecific::get_param_vocab());
+
+	ret.push_back(ParamDesc("layer",Param::TYPE_LAYER)
+		.set_local_name(_("Layer"))
+		.set_desc(_("Layer to be lowered"))
+		.set_supports_multiple()
+	);
+
+	return ret;
+}
+
+bool
+Action::LayerLowerMax::is_candidate(const ParamList &x)
+{
+	if(!candidate_check(get_param_vocab(),x))
```

Comment:
please let's try to use current voted style... spaces before `(` (when related to `if`, `for`, `while`) and after `,`

Reply:
Yes it's actually the style I use personally too. This is mostly because I used the layerraise/lower actions as sort of templates to this action so the style carried over from there. will do.